transport/src/main/java/io/netty/channel/FileChannelBasedSlicedFileRegion.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * A derived {@link FileChannelBasedFileRegion} which shares reference counting with its parent.
+ */
+final class FileChannelBasedSlicedFileRegion extends FileChannelBasedFileRegion {
+
+    private final FileChannelBasedFileRegion parent;
+
+    private final long offset;
+
+    private final long count;
+
+    public FileChannelBasedSlicedFileRegion(FileChannelBasedFileRegion parent, long index,
+            long count) {
+        if (index < 0 || index > parent.count() - count) {
+            throw new IndexOutOfBoundsException(parent + ".slice(" + index + ", " + count + ')');
+        }
+
+        if (parent instanceof FileChannelBasedSlicedFileRegion) {
+            FileChannelBasedSlicedFileRegion sliced = (FileChannelBasedSlicedFileRegion) parent;
+            this.parent = sliced.parent;
+            this.offset = sliced.offset + index;
+        } else {
+            this.parent = parent;
+            this.offset = index;
+        }
+        this.count = count;
+    }
+
+    @Override
+    public long position() {
+        return parent.position() + offset;
+    }
+
+    @Override
+    public long transfered() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long transferred() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long count() {
+        return count;
+    }
+
+    @Override
+    public long transferTo(WritableByteChannel target, long position) throws IOException {
+        throw new UnsupportedOperationException();
```

Comment:
For now `SliceableFileRegion` just extends `FileRegion`, the intention here is to move the methods in this interface up to `FileRegion` at the next minor release. The old `transferTo` and `trnsferred` methods are marked as deprecated and some new methods are provided. You can either use the old methods, or the new methods, but you can not use them both at the same time. So you mean it is better to declare a completely new interface and remove the old `FileRegion` interface in the future?